### ISSUE TITLE
Device to detect what state an item is in

### ISSUE THREAD

**Issue by agent-1t** [studio-minus/ppg-bugs-and-requests#1948]
**Feature description**
A device which detects what state an item is in and outputs an activation signal when the item's state changes.

**Feature purpose**
You could make a defiblirator which only revives people when they flatline, a missile launcher that only fires when it detects something, etc.
**Additional context**
I wanted to make an automatic surface to air missile system but couldn't use detectors because they weren't precise enough.

**Comment by DindinYT37** [comment]
You know detectors exist, right?

**Comment by bibbybabby** [comment]
> You know detectors exist, right?

**Additional context**
I wanted to make an automatic surface to air missile system _but couldn't use detectors because they weren't precise enough._

**Comment by DindinYT37** [comment]
> > You know detectors exist, right?
> 
> **Additional context**
> I wanted to make an automatic surface to air missile system _but couldn't use detectors because they weren't precise enough._

Yeah but he is this request supposed to fix that?

**Comment by agent-1t** [comment]
> Yeah but how is this request supposed to solve that?

It would be able to fire the guided missiles at the exact moment they lock on instead of a few miliseconds off, and life detectors also aren't precise enough, only triggering when the human is fully dead. Also another example I should've mentioned is you could create failsafe systems by detecting if a component is broken or not.

**Comment by DindinYT37** [comment]
> > Yeah but how is this request supposed to solve that?
> 
> It would be able to fire the guided missiles at the exact moment they lock on instead of a few miliseconds off, and life detectors also aren't precise enough, only triggering when the human is fully dead. Also another example I should've mentioned is you could create failsafe systems by detecting if a component is broken or not.

Your issue has nothing to do with the detectors, it's simply the propagation delay.

**Comment by agent-1t** [comment]
> Your issue has nothing to do with the detectors, it's simply the propagation delay.

Fair enough, I think the second two issues would still be solved with it though.
